lib/active_support/cache/database_store/model.rb

```diff
@@ -15,7 +15,13 @@ def self.truncate!
         scope :created_before, ->(date = 1.month.ago) { where(arel_table[:created_at].lt(date)) }
 
         def self.namespaced(namespace)
-          where(arel_table[:key].matches("#{namespace}:%"))
+          case ActiveRecord::Base.connection.adapter_name
+          when 'PostgreSQL'
+            ifx = Arel::Nodes::InfixOperation.new('IN', Arel::Nodes.build_quoted(namespace), arel_table[:key])
+            where(Arel::Nodes::NamedFunction.new('POSITION', [ifx]).eq(1))
```

Comment:
**issue:** I might be mistaken, I don't have big exposure with MySQL and MariaDB.

But this AREL query could be used for MariaDB/MySQL case as well. 

Seems like `IN`/`POSITION` is supported by both. I couldn't find mentions about `IN` in documentation, but I found comments in various pages where people mention that it works.

Sqlite doesn't seem to support `POSITION`, though.